### PR TITLE
Update docs for Flux v0.26.0

### PR DESCRIPTION
**Pull request by stefanprodan**
Changes:
- set min required version to Kubernetes 1.20.6 in install doc
- add Multi-tenancy lockdown section to install docs 
- add Signed Container Images section to the security doc
- add Software Bill of Materials section to the security doc
- add Pod Security Standard section to the security doc

Part of: https://github.com/fluxcd/flux2/issues/2308

### CODE REVIEW

File: content/en/docs/security/_index.md [hunk 1]
```diff
@@ -15,17 +17,58 @@ Please follow [this GitHub issue](https://github.com/fluxcd/website/issues/598)
 
 Flux has a multi-component design, and integrates with many other systems.
 
-This document outlines an overview of security considerations for Flux components, project processes, artifacts, as well as Flux configurable options and what they enable for both Kubernetes cluster and external system security.
+This document outlines an overview of security considerations for Flux components,
+project processes, artifacts, as well as Flux configurable options and
+what they enable for both Kubernetes cluster and external system security.
```
Comment (makkes) [inline content/en/docs/security/_index.md:22]: I'm not sold on hard-wrapping lines in Markdown but that's just a personal preference as I find it much easier to diff changes when paragraphs are not hard-wrapped.

File: content/en/docs/security/_index.md [hunk 2]
```diff
@@ -15,17 +17,58 @@ Please follow [this GitHub issue](https://github.com/fluxcd/website/issues/598)
...
 
-See our [security processes document](/security) for vulnerability reporting, handling, and disclosure of information for the Flux project and community.
+See our [security processes document](/security) for vulnerability reporting, handling,
+and disclosure of information for the Flux project and community.
+
+## Signed container images
+
+The Flux CLI and the controllers' images are signed using [Sigstore](https://www.sigstore.dev/) Consign and GitHub OIDC.
```
Comment (makkes) [inline content/en/docs/security/_index.md:29]:
```suggestion
The Flux CLI and the controllers' images are signed using [Sigstore](https://www.sigstore.dev/) Cosign and GitHub OIDC.
```

**Review by makkes — COMMENTED** [review]
I tested the multi-tenancy lockdown kustomization and it works just fine. The rest is just nits.

**Review by dholbach — APPROVED** [review]
:rocket:

**Pull request merged**